lib/realtime/tenants/connect.ex

```diff
@@ -366,6 +398,21 @@ defmodule Realtime.Tenants.Connect do
     with tenant <- Tenants.Cache.get_tenant_by_external_id(tenant_id),
          :ok <- tenant_suspended?(tenant),
          {:ok, node} <- Realtime.Nodes.get_node_for_tenant(tenant) do
+      opts =
+        if node == node() do
+          # Tracing context is passed for local calls
+          # This is necessary because :erpc will spawn a process when timeout is not :infinity
+          span_ctx = Tracer.start_span("database.connect")
+          ctx = OpenTelemetry.Ctx.get_current()
+          [otel: %{span_ctx: span_ctx, ctx: ctx}] ++ opts
+        else
+          # For actual remote calls we pass whatever context propagation we have available
+          case :otel_propagator_text_map.inject([]) do
+            [] -> []
+            otel_propagation -> [otel: otel_propagation]
+          end
+        end
```

Comment:
are we able to move this outside of the Connect code somehow? Connect it's already quite complex and this adds quite a lot of readability load to this code